### DISCUSSION TITLE
Load data async when use useExpanded and render custom subheaders and subrows?

### DISCUSSION THREAD

**Discussion by V-Honcharov** [TanStack/table#2084] [General]
Is it possible to render in expended body subheader and another cell? For example, I have table:
     |ID|      Name   |Status| <- main header
👉|1  | mainItem1|Success| <- main row
         |subItemID|subItemName| subItemStatus| subItemState| <- subheader
         |21321321|    subitem      |      Success    |   Deleted   | <- subrow
And is it possible to download sub rows only on expand button click?

**Comment by sanchitos** [comment]
Take a look here [https://github.com/TanStack/table/blob/main/examples/react/sub-components/src/main.tsx](https://github.com/TanStack/table/blob/main/examples/react/sub-components/src/main.tsx)
